src/coreclr/jit/hwintrinsiccodegenxarch.cpp

```diff
@@ -1968,6 +1969,59 @@ void CodeGen::genBaseIntrinsic(GenTreeHWIntrinsic* node, insOpts instOptions)
             break;
         }
 
+        case NI_Vector128_WithElement:
+        case NI_Vector256_WithElement:
+        case NI_Vector512_WithElement:
+        {
+            // Optimize the case where op2 is not a constant.
+            if (!op2->OperIsConst())
```

Comment:
Should we add a path, like GetElement has, that optimizes for `!op1->isUsedFromReg()`, that is the case where `op1` already exists in memory and we're trying to replace the nth element?

Reply:
Actually, nevermind, that would only be valid in the case we have some `Load(addr).WithElement(n, value).Store(addr)`, so nothing to do here.

Reply:
Might be an additional optimization to look at in the future, but not necessary for this PR.